package.json

```diff
@@ -7,13 +7,13 @@
     "node": ">=18 <21"
   },
   "browserslist": "> 0.25%, not dead",
-  "main": "frontend/js/index.js",
+  "main": "frontend/js/index.tsx",
   "scripts": {
     "test": "jest",
     "test:watch": "npm test -- --watch",
     "test:update": "npm test -- --u",
     "dev": "webpack serve --mode=development --hot",
-    "build": "NODE_ENV=production webpack --progress --bail --mode=production",
+    "build": "NODE_ENV=production tsc && webpack --progress --bail --mode=production",
```

Comment:
We need to add tsc to pre-commit as well. Can we add on `lint`?

Reply:
I added another script because `lint` is already used in the `pre-commit` for the `eslint` check.